### PR TITLE
Stability tests: add retry logic

### PR DESCRIPTION
**Pull request by matthewfala**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/firelens-stability/lib/cloud/ecs.ts [hunk 1]
```diff
@@ -41,86 +46,229 @@ async function processEcsTestTaskQueue() {
...
+    for (let i = 0; i < taskCount; i += Math.min(10, taskCount - i)) {
+        const count = Math.min(10, taskCount - i);
+        let result: PromiseResult<AWS.ECS.RunTaskResponse, AWS.AWSError>;
         try {
-            taskDefinitionArn = (await ecs.registerTaskDefinition(taskDefinition).promise()).taskDefinition!.taskDefinitionArn;
+            result = await ecs.runTask({
+            cluster: testCase.config.cluster,
+            taskDefinition: taskDefinitionArn!,
+            count: count,
+            launchType: "FARGATE",
```
Comment (PettitWesley) [inline apps/firelens-stability/lib/cloud/ecs.ts:218]: do we want to harcode the launchType??

Comment (PettitWesley) [reply apps/firelens-stability/lib/cloud/ecs.ts:218]: if we parametrize it from start its easier to change in future.

I think we will stick with Fargate long term tho, so its fine.

File: apps/firelens-stability/lib/cloud/ecs.ts [hunk 2]
```diff
@@ -41,86 +46,229 @@ async function processEcsTestTaskQueue() {
...
+    let taskArns = ecsTestTask.executionRecord.taskArns;
+    if (endedTasksCount !== 0) {
+        console.log(`  🔁 ${testCase.managed.collectionName}/${testCase.managed.suiteName}/${testCase.managed.caseName}: Retrying failed tasks`);
+        const taskCount = endedTasksCount;
+        let newTasksLaunched = await launchECSTasks(testCase, taskCount, ecs, ecsTestTask.taskDefinitionArn);
+        taskArns = [...startingTasks, ...runningTasks, ...newTasksLaunched];
+    }
+
+    /* Add back to queue for validation */
+    addECSTaskToQueue(10000, 20000, {
```
Comment (PettitWesley) [inline apps/firelens-stability/lib/cloud/ecs.ts:165]: huge magic numbers: can these be constants and why is hte base so big?

Comment (PettitWesley) [reply apps/firelens-stability/lib/cloud/ecs.ts:165]: are these milliseconds?

Comment (matthewfala) [reply apps/firelens-stability/lib/cloud/ecs.ts]: Updated.

**Pull request merged**